optuna/samplers/cmaes/sampler.py

```diff
@@ -0,0 +1,351 @@
+            :func:`~optuna.samplers.intersection_search_space()`.
+
+            If :obj:`None` is specified, :class:`~optuna.samplers.RandomSampler` is used
+            as the default.
+
+            .. seealso::
+                :class:`optuna.samplers` module provides built-in independent samplers
+                such as :class:`~optuna.samplers.RandomSampler` and
+                :class:`~optuna.samplers.TPESampler`.
+
+        warn_independent_sampling:
+            If this is :obj:`True`, a warning message is emitted when
+            the value of a parameter is sampled by using an independent sampler.
+
+            Note that the parameters of the first trial in a study are always sampled
+            via an independent sampler, so no warning messages are emitted in this case.
+    """
+
+    def __init__(
+            self,
+            x0: Optional[Dict[str, Any]] = None,
+            sigma0: Optional[float] = None,
+            n_startup_trials: int = 1,
+            independent_sampler: Optional[BaseSampler] = None,
+            warn_independent_sampling: bool = True,
+            seed: Optional[int] = None,
+    ) -> None:
+
+        self._x0 = x0
+        self._sigma0 = sigma0
+        self._independent_sampler = (
+            independent_sampler or optuna.samplers.RandomSampler(seed=seed)
+        )
+        self._n_startup_trials = n_startup_trials
+        self._warn_independent_sampling = warn_independent_sampling
+        self._logger = optuna.logging.get_logger(__name__)
+        self._cma_rng = np.random.RandomState(seed)
+
+    def infer_relative_search_space(
+            self, study: 'optuna.Study', trial: 'optuna.structs.FrozenTrial',
+    ) -> Dict[str, BaseDistribution]:
+
+        # Import here to avoid circular imports without breaking a backward compatibility.
+        from optuna.samplers import intersection_search_space
+
+        search_space = {}  # type: Dict[str, BaseDistribution]
+        for name, distribution in intersection_search_space(study).items():
+            if distribution.single():
+                # `cma` cannot handle distributions that contain just a single value, so we skip
+                # them. Note that the parameter values for such distributions are sampled in
+                # `Trial`.
+                continue
+
+            if not isinstance(
+                    distribution,
+                    (
+                        optuna.distributions.UniformDistribution,
+                        optuna.distributions.LogUniformDistribution,
+                        optuna.distributions.DiscreteUniformDistribution,
+                        optuna.distributions.IntUniformDistribution,
+                    ),
+            ):
+                # Categorical distribution is unsupported.
+                continue
+            search_space[name] = distribution
+
+        if len(search_space) < 2:
+            self._logger.info("`CmaEsSampler` only supports two or more dimensional continuous "
```

Comment:
I think that it's better to emit this message only when there are one or more finished (not running) trials. Currently, it seems the first trial always emits this message.